std/socket.d

```diff
@@ -1993,7 +2012,9 @@ static if (is(sockaddr_un))
 
         @property string path() @trusted const pure
         {
-            return to!string(cast(const(char)*)sun.sun_path.ptr);
+            return (cast(const(char)*) sun.sun_path.ptr)[0 ..
+                _nameLen - sockaddr_un.init.sun_path.offsetof -
+                (sun.sun_path.ptr[0] ? 1 : 0)].idup;
         }
```

Comment:
The excluded byte is the null byte terminating a pathname address (added in the constructor to make it a C string).
For abstract address (`sun.sun_path.ptr[0] == 0`), nothing is excluded.